hydromt/components/vector.py

```diff
@@ -162,20 +169,25 @@ def read_vector(
             Additional keyword arguments that are passed to the `read_nc`
             function.
         """
-        self._assert_read_mode()
+        self._root._assert_read_mode()
         self._initialize_vector(skip_read=True)
         if fn is not None:
             # Disable lazy loading of data
             # to avoid issues with reading object dtype data
             if "chunks" not in kwargs:
                 kwargs["chunks"] = None
-            ds = xr.merge(self.read_nc(fn, **kwargs).values())
+            ds = xr.merge(
+                read_nc(
+                    fn, root=self._root.path, logger=self._logger, **kwargs
+                ).values()
+            )
             # check if ds is empty (default fn has a value)
             if len(ds.sizes) == 0:
                 fn = None
-        if fn_geom is not None and isfile(join(self.root.path, fn_geom)):
-            gdf = gpd.read_file(join(self.root.path, fn_geom))
+        if fn_geom is not None and isfile(join(self._root.path, fn_geom)):
+            gdf = gpd.read_file(join(self._root.path, fn_geom))
             # geom + netcdf data
+            # TODO: What if ds is None?
```

Comment:
Isn't it taken into account ? lines 185-186 makes sure ds is not empty and I assume read_nc would complain if file not found. Extra I think that the default for data_vars is None anyway.